### PR TITLE
smartmontools: 7.0 -> 7.1 and devicedb updated to latest

### PR DESCRIPTION
**Pull request by Frostman**
###### Motivation for this change

https://www.smartmontools.org/browser/tags/RELEASE_7_1/smartmontools/NEWS

Currently, we have 1+ year old devicedb.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---
This change is [Reviewable](https://reviewable.io/reviews/nixos/nixpkgs/81099)

### CODE REVIEW

File: pkgs/tools/system/smartmontools/default.nix [hunk 1]
```diff
@@ -2,13 +2,13 @@
 , IOKit ? null , ApplicationServices ? null }:
 
 let
-  version = "7.0";
+  version = "7.1";
 
-  dbrev = "4883";
-  drivedbBranch = "RELEASE_${builtins.replaceStrings ["."] ["_"] version}_DRIVEDB";
+  dbrev = "5033";
+  drivedbBranch = "RELEASE_7_0_DRIVEDB";
```
Comment (Frostman) [inline pkgs/tools/system/smartmontools/default.nix]: Note: there is no dedicated branch for 7.1. The 7.0 branch is supposed to be used for both 7.0 and 7.1.

**Comment by Frostman** [comment]
cc @dtzWill @xeji @andir

**Comment by andir** [comment]
@worldofpeace any objections against including this in 20.03? It seems like a good addition.

**Comment by Frostman** [comment]
I've backported and tested it #81423

**Comment by dtzWill** [comment]
Belated :+1:, thanks!

**Pull request merged**